lambda/matching.ml

```diff
@@ -1048,108 +1048,44 @@ let pat_as_constr = function
   | { pat_desc = Tpat_construct (_, cstr, _) } -> cstr
   | _ -> fatal_error "Matching.pat_as_constr"
 
-let group_const_int p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_int _) -> true
-  | _ -> false
-
-let group_const_char p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_char _) -> true
-  | _ -> false
-
-let group_const_string p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_string _) -> true
-  | _ -> false
-
-let group_const_float p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_float _) -> true
-  | _ -> false
-
-let group_const_int32 p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_int32 _) -> true
-  | _ -> false
-
-let group_const_int64 p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_int64 _) -> true
-  | _ -> false
-
-let group_const_nativeint p =
-  match Pattern_head.desc (Simple.head p) with
-  | Constant (Const_nativeint _) -> true
-  | _ -> false
-
-and group_constructor p =
-  match Pattern_head.desc (Simple.head p) with
-  | Construct _ -> true
-  | _ -> false
-
-and group_same_constructor tag p =
-  match Pattern_head.desc (Simple.head p) with
-  | Construct cstr -> Types.equal_tag tag cstr.cstr_tag
-  | _ -> false
-
-and group_variant p =
-  match Pattern_head.desc (Simple.head p) with
-  | Variant _ -> true
-  | _ -> false
-
 and group_var p =
```

Comment:
I am not sure if `group_var` needs to be grouped with `pat_as_constructor`.  It might be also better to rename the function now that it is has no relationship with `can_group`?

Reply:
I pushed a minor commit to rename `group_var` into `simple_omega_like` -- we already have an `omega_like` function that does morally the same thing on general patterns.